cmd/gitsign-attest/main.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 The Sigstore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/sigstore/gitsign/cmd/gitsign-attest/internal/attest"
+)
+
+var (
+	tree    = flag.Bool("t", false, "sign tree instead of commit")
+	path    = flag.String("f", "", "file to attest")
+	attType = flag.String("type", "", "attestation type")
```

Comment:
Is `type` here a fully freeform string? Is there any validation we should do to prevent `-type sdpx` typos?

Reply:
Done.